### ISSUE TITLE
automatix broken due to new rsemgr.upload function

### ISSUE THREAD

**Issue by vingar** [rucio/rucio#1582]
Motivation
----------

`rucio/tests/daemons/Automatix.py` line 86 is expecting two values while only one dict is returned generating a too many values to unpack error.

guilty line:
 `global_status, ret = rsemgr.upload(rse_info, lfns=lfns, source_dir=source_dir)`

while it receives instead for example :

`{0: True, 1: {u'tests:RAW.2b409202459145d2a02aad377438c897': True, u'tests:RAW.4cd16c00832b43fe8d9d85a96e13211e': True}, 'success': True, 'pfn': u'root://eosatlas.cern.ch:1094//eos/atlas/atlasdatadisk/ruciotest/rucio/tests/30/0a/RAW.4cd16c00832b43fe8d9d85a96e13211e'}`

Modification
------------

Fix the functional test!

**Comment by vingar** [comment]
one more error came:
```
2018-09-28 08:42:34,153	7986	DEBUG	Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/rucio/tests/daemons/Automatix.py", line 115, in upload
    client.add_dataset(scope=dsn['scope'], name=dsn['name'], rules=[{'account': account, 'copies': 1, 'rse_expression': rse, 'grouping': 'DATASET', 'activity': 'Functional Test'}], meta=meta, lifetime=dataset_lifetime)
  File "/usr/lib/python2.7/site-packages/rucio/client/didclient.py", line 149, in add_dataset
    lifetime=lifetime, dids=files, rse=rse)
  File "/usr/lib/python2.7/site-packages/rucio/client/didclient.py", line 119, in add_did
    raise exc_cls(exc_msg)
RucioException: An unknown exception occurred.
Details: [u'(cx_Oracle.DatabaseError) ORA-20101: Primary key constraint DELETED_DIDS_PK violated\nORA-06512: at "ATLAS_RUCIO.CHECK_DID_UNIQUENESS", line 12\nORA-04088: error during execution of trigger \'ATLAS_RUCIO.CHECK_DID_UNIQUENESS\'\n']
```

**Comment by cserf** [comment]
I'm not sure what is the best thing to do. The patch that modifies the output of the upload method (
https://github.com/rucio/rucio/commit/c839a8ee4fb1399116853f532ef9e7d5a288af18 has comment "rsemanager: dirty dirty fix of upload method return - #1487"). So I would like to know if the dirty fix will stay, or will it be replaced by something cleaner.

**Comment by bari12** [comment]
Need to check with @TomasJavurek and @TWAtGH
